cmd/charm-ingest/whitelist_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the GPLv3, see LICENCE file for details.
+
+package main
+
+import (
+	"strings"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/juju/charmstore-client/internal/ingest"
+	"gopkg.in/juju/charmrepo.v4/csclient/params"
+)
+
+var whiteListTests = []struct {
+	tag         string
+	whitelist   string
+	expect      []ingest.WhitelistEntity
+	expectError string
+}{{
+	tag: "valid_whitelist",
+	whitelist: `
+wordpress stable
+elasticsearch-30 beta stable
+
+trusty/redis-0 unpublished
+mongodb
+`,
+	expect: []ingest.WhitelistEntity{
```

Comment:
If you put the two opening `{{` on the same line, and each divider is `}, {`, you can save a level of indentation all the way down.